### ISSUE TITLE
[Xedra Evolved] Dreamforged recipe issues

### ISSUE THREAD

**Issue by rty275** [CleverRaven/Cataclysm-DDA#66613]
### Describe the bug

The dreamforged arrow recipe is called "dreamforged long sword" despite having the correct description, and generates a dreamforged longsword in addition to whatever arrows were crafted.

The dreamforged arm guards recipe is called "dreamforged plate armor", and produces poor fitting equipment.

The dreamforged full helmet recipe produces poor fitting equipment.

### Attach save file

[Innawoods_copy-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/11930068/Innawoods_copy-trimmed.tar.gz)

### Steps to reproduce

1. Develop dreamsmith abilities with inquiry (Already complete in attached save)
2. Craft "dreamforged long sword" for far less materials than one might expect and receive arrows in addition to a longsword
3. Craft "dreamforged plate armor" and be surprised when poorly fitted arm guards come out
4. Craft dreamforged full helmet and be first relieved that it produces the correct thing, then annoyed that it is poorly fitted.

### Expected behavior

Custom crafted armor made of dreams should fit well. Also, long swords should be long swords, not arrows.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3086 (22H2)
- Game Version: 593bafd [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Magiclysm [magiclysm],
    Aftershock [aftershock],
    Bionic Professions [package_bionic_professions],
    Blaze Industries [blazeindustries],
    Xedra Evolved [xedra_evolved],
    Mind Over Matter [mindovermatter],
    DinoMod [DinoMod],
    Mining Mod [Mining_Mod],
    Mythical Martial Arts [MMA],
    Bombastic Perks [bombastic_perks],
    Extra Mutated Scenarios [extra_mut_scens],
    Bionic Slots [cbm_slots],
    Magiclysm No Class Limit [magiclysm_no_class_limit],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Tamable Wildlife [Tamable_Wildlife],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Disable NPC Needs [no_npc_food],
    Minimod: No Skill Rust [minimod_no_rust],
    Innawood [innawood]
]

### Additional context

Hopefully the skill rust minimod isn't required to reproduce, but I guess if it is then I know a mod to cut from my usual repertoire. 

It's possible that other dreamforged recipes are acting up too, but these are the ones I've found so far and the research process takes a bit of time to complete.

**Comment by GuardianDll** [comment]
> The dreamforged arrow recipe is called "dreamforged long sword" despite having the correct description, and generates a dreamforged longsword in addition to whatever arrows were crafted.

the reason is that `arrow_dreamforged` recipe copies `dreamforged_longsword_ench`, which incorrectly copies values - it copies `name`, `description` and `result_eocs` (which is hillarious, because code itself doesn't allow to use this fields if you have `result`).

\> poor fitting

and that's something i mess up

Fix on the way

**Comment by rty275** [comment]
Copy_from lets you do things that directly coding it doesn't? That's silly and amusing. Thanks for the quick response!

**Comment by GuardianDll** [comment]
Copy_from is form of inheritance, it allows to simplify json by marking one block as parent, and write the child blocks, omitting the parts you don't need (because why i need to write category, subcategory and skill_used thirty times in a row, if i can write it only once)

**Comment by rty275** [comment]
Oh for sure, it totally makes sense to use it, it's just funny that it lets you circumvent little limitations like not normally letting you use those fields
